NFS storage documentation (#835)
netgroup-polito/CrownLabs: operators/README.md

When the snapshot creation process successfully terminates, the docker registry will contain a new VM image with the exact copy of the target persistent VM at the moment of the snapshot creation. Note that before being able to create a new VM instance with that image, you should first create a new Template with the newly uploaded image.

### Personal storage

The Instance Operator, can attach to the instance the user's personal storage using the parameters stored in the `mydrive-info` secret created by the Tenant Operator. 

> A flag named `MountMyDriveVolume` in the `Environment` inside the `Template` of the `Instance` resource to reconcile is used to decide whether to mount or not the personal storage.

Reviewer: I cannot understand the sentence. What does it mean "to reconcile" here?

Author: The operation that the Instance Operator does where it changes the Instance status from the actual state to the desired state is called reconciliation.

Reviewer: I know what "reconciliation" is :-) , but the sentence looks weird to me... can you please rephrase to make it more clear? I cannot do myself, as I don't understand what you wanted to say :-(

Author: I was trying to say that the flag is found inside the Environment that is inside the Instance resource that is being reconciled by the operator.
I could write the path to the flag like: `instance.environments\[\].MountMyDriveVolume` but I don't know if that is clear enough.
Another way could be:
```
With the `MountMyDriveVolume` flag inside each `Environment` found in a Instance Resource the Instance Operator mounts or not the user's personal storage inside that `Environment`
```

Reviewer 2: That last version should be more clear

Reviewer 2: I made a sentence based on @matteorosani suggestion

This operation is performed both for Containers and VirtualMachines.

> In the Container case the user's personal storage is attached as a NFS volume to the `Pod` and then mounted in the contanier environment with a `VolumeMount`.

Reviewer: ```suggestion
For containers, the user's personal storage is attached as a NFS volume to the `Pod` and then mounted in the container environment with a `VolumeMount`.
```

> In the VirtualMachine case the user's personal storage is attached by the VM itself. Using `cloud-init` we add the mount point to the VM's `/etc/fstab` file and the machine tries to mount it using the NFS filesystem.\

Reviewer: ```suggestion
For Virtual Machines, the user's personal storage is attached by the VM itself: `cloud-init` is used to add the mount point to the VM's `/etc/fstab` file and the machine tries to mount it using the NFS filesystem.\
```

  - after every update on the cluster resources managed by the operator itself.
  - periodically, (i.e. in a range between 1 and 2 hours since the last update)
- When performing the actions, the operator utilizes a `fail-fast:false` strategy. Hence, if an action fails (e.g. due to Keycloak being temporarily offline), the operator does not stop and tries to execute all the other independent actions.

> - The PVCs for the users personal storage are created all into one namespace, specified by the `--mydrive-pvcs-namespace` parameter of the operator, this has been done to prevent the loss of the user data after the deletion of the user account.

Reviewer: ```suggestion
- PVCs for the users personal storage are all created in the same namespace, specified by the `--mydrive-pvcs-namespace` parameter of the operator, and different from the namespace of the user.
This prevents the loss of the user data after the deletion of the user account, which removes also the namespace of the user itself and all the resources therein.
```
